src/js/services/conversation-service.js

```diff
@@ -95,7 +96,9 @@ export function uploadImage(file) {
             return core().conversations.uploadImage(user._id, blob, {
                 role: 'appUser'
             }).then((response) => {
-                store.dispatch(setConversation(response.conversation));
+                // Merge messages from response with messages from store state
+                const conversation = mergeMessages(response, store.getState().conversation.messages, message);
```

Comment:
I would move that logic in the reducer. That way it would get executed everytime it gets dispatched,